<a href="https://colab.research.google.com/github/harsha1067/ASSIGNMENTS/blob/main/Assignment_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? 

Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

Assume

Null Hypothesis, H0: μ1 = μ2

Alternate Hypothesis, H1: μ1 != μ2

In [ ]:
import pandas as pd 
import numpy as np 
from scipy import stats 

In [ ]:
data = pd.read_csv('Cutlets.csv')
data

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [ ]:
df1 = pd.Series(data.iloc[:,0])
df1

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [ ]:
df2 = pd.Series(data.iloc[:,1])
df2

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [ ]:
pvalue = stats.ttest_ind(df1, df2)
pvalue

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

pvalue is greater than alpha value.

0.4722 > 0.05

Therfore, accept null hypothesis.

2)A hospital wants to determine whether there is any differnece in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.

Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

Assume

Null Hypothesis, H0: μ1 = μ2

Alternate Hypothesis, H1: μ1 != μ2

In [ ]:
import pandas as pd 
from scipy import stats

In [ ]:
data = pd.read_csv("LabTAT.csv")
data

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [ ]:
pvalue = stats.f_oneway(data.iloc[:,0], data.iloc[:,1], data.iloc[:,2], data.iloc[:,3])
pvalue

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

pvalue = 2.12

alpha = 0.05

pvalue > alpha

i.e., 2.12 > 0.05

Therefore, accept null hypothesis.


3)Sales of products in four different regions is tabulated for males and females. Find if male-female buyer ratios are similar across regions.






Assume

Null Hypothesis, H0: μ1 = μ2

Alternate Hypthesis, H1: μ1 != μ2

In [1]:
import pandas as pd 
import numpy as np 
from scipy import stats 
from scipy.stats import chi2_contingency 
from scipy.stats import chi2

In [2]:
data = pd.read_csv('BuyerRatio.csv')
data

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [5]:
df = data.iloc[:,1:6]
df

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [6]:
df.values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [8]:
df1 = stats.chi2_contingency(df)
df1

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

pvalue = 0.66

alpha = 0.05

Since, pvalue > alpha i.e., 0.66 > 0.05

Therefore, Null Hypothesis is accepted.

4) TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by center. Please analyze the data at 5% significance level and help the manager draw appropriate inferences.

Assume

Null Hypothesis, H0: μ1 = μ2

Alternate Hypothesis, H1: μ1 != μ2

In [15]:
import pandas as pd
from scipy import stats 
from scipy.stats import chi2_contingency 
from scipy.stats import chi2 

In [16]:
data = pd.read_csv('Costomer+OrderForm.csv')
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [17]:
print(data['Phillippines'].value_counts(),data['Indonesia'].value_counts(),data['Malta'].value_counts(),data['India'].value_counts())

Error Free    271
Defective      29
Name: Phillippines, dtype: int64 Error Free    267
Defective      33
Name: Indonesia, dtype: int64 Error Free    269
Defective      31
Name: Malta, dtype: int64 Error Free    280
Defective      20
Name: India, dtype: int64


In [18]:
df = ([271, 267, 269, 280], [29, 33, 31, 20])
df

([271, 267, 269, 280], [29, 33, 31, 20])

In [20]:
stat, p, dof, expected = chi2_contingency([[271,267,269,280],[29,33,31,20]])

In [21]:
stat

3.858960685820355

In [22]:
p 

0.2771020991233135

p_value = 0.277

alpha = 0.05

Since, p_value > alpha i.e., 0.277 > 0.05

Therefore, Null Hypothesis is accepted.